serialization/BSON.ksy

```diff
@@ -0,0 +1,165 @@
+meta:
+  id: bson
+  file-extension: bson
+  endian: le
+  license: CC0
+doc: >
+  BSON, short for Binary JSON, is a binary-encoded serialization of JSON-like documents. Like JSON, BSON supports the embedding of documents and arrays within other documents and arrays. BSON also contains extensions that allow representation of data types that are not part of the JSON spec. For example, BSON has a Date type and a BinData type.
+  BSON can be compared to binary interchange formats, like Protocol Buffers. BSON is more "schemaless" than Protocol Buffers, which can give it an advantage in flexibility but also a slight disadvantage in space efficiency (BSON has overhead for field names within the serialized data).
+  BSON was designed to have the following three characteristics:
+    * Lightweight. Keeping spatial overhead to a minimum is important for any data representation format, especially when used over the network.
+    * Traversable. BSON is designed to be traversed easily. This is a vital property in its role as the primary data representation for MongoDB.
+    * Efficient. Encoding data to BSON and decoding from BSON can be performed very quickly in most languages due to the use of C data types.
+seq:
+  - id: doc
+    type: document
+types:
+  code_with_scope:
+    doc: ""
+    seq:
+      - id: id
+        type: s4
+      - id: source
+        type: string
+      - id: scope
+        type: document
+        doc: "mapping from identifiers to values, representing the scope in which the string should be evaluated."
+  document:
```

Comment:
In fact in the spec this comment is for top-level doc. I'll fix this, thank you.